### ISSUE TITLE
Add okio or kotlinx-io Path conversion extensions

### ISSUE THREAD

**Issue by mgroth0** [ajalt/clikt#492]
Title says it all. Using something like [kotlinx-io](https://github.com/Kotlin/kotlinx-io) to support even a subset of the path features in native would be excellent

**Comment by JakeWharton** [comment]
Since that library is not stable, you're better off doing a manual convert for now. I suspect once it does go stable it will be supported.

You could also use Okio's multiplatform file system. It is stable, but I wouldn't expect Clikt to take it as a dependency for built-in support. Example:

https://github.com/JakeWharton/video-swatch/blob/trunk/src/commonMain/kotlin/com/jakewharton/videoswatch/main.kt#L69-L70

**Comment by ajalt** [comment]
I don't generally have library-specific extensions since calling `.convert { it.toWhatever }` seems easy enough. But since paths have some extra validation functionality like `canBeFolder`, it might make sense to provide that once kotlinx-io is stable.

**Comment by Wavesonics** [comment]
makes sense, might be nice to add an example to the docs where it says that file and path are only available on the jvm, and point people to this solution.

**Comment by ajalt** [comment]
The docs on [multiplatform support][1] already mention that they're JVM-only.

[1]: https://ajalt.github.io/clikt/advanced/#multiplatform-support

**Comment by Wavesonics** [comment]
Right, I'm suggesting it mention that for other targets you could use Okio and add a `.convert()`
